### PR TITLE
Fixed Issue #10931 Complement of S.Integers with itself and S.Reals.

### PR DESCRIPTION
**Pull request by nc1337**
Now this is evaluating right.

```
In [2]: S.Integers - S.Integers
Out[2]: EmptySet()

In [3]: S.Integers - S.Reals
Out[3]: EmptySet()
```

If something is wrong then please guide me.

### CODE REVIEW

**Comment by nc95** [comment]
Someone review this PR.

File: sympy/sets/sets.py [hunk 1]
```diff
@@ -1619,7 +1619,8 @@ def reduce(A, B):
         Simplify a :class:`Complement`.
 
         """
-        if B == S.UniversalSet:
+        # Fixed Issue 10931
+        if B == S.UniversalSet or B == A or (A == S.Integers and B == S.Reals):
```
Comment (smichr) [inline sympy/sets/sets.py]: Could this just be `if A.is_subset(B):`?

Comment (nc95) [inline sympy/sets/sets.py]: Yeah, that's right @smichr. Thanks

**Comment by nc95** [comment]
@smich It seems that Intersection between the ```ImageSets``` is not working fine as it can't be iterated.This is why the check if failing.

**Pull request closed without merging**